comp/api/api/apiimpl/internal/config/endpoint.go

```diff
@@ -36,18 +37,44 @@ type configEndpoint struct {
 	errorsExpvar       expvar.Map
 }
 
-func (c *configEndpoint) serveHTTP(w http.ResponseWriter, r *http.Request) {
-	body, statusCode, err := c.getConfigValueAsJSON(r)
-	if err != nil {
-		http.Error(w, err.Error(), statusCode)
+func (c *configEndpoint) getConfigValueHandler(w http.ResponseWriter, r *http.Request) {
+	vars := gorilla.Vars(r)
+	path := vars["path"]
+	// escape in case it contains html special characters that would be unsafe to include as is in a response
+	// all valid config paths won't contain such characters so for a valid request this is a no-op
+	path = html.EscapeString(path)
+
+	if _, ok := c.authorizedConfigPaths[path]; !ok {
+		c.unauthorizedExpvar.Add(path, 1)
+		log.Warnf("config endpoint received a request from '%s' for config '%s' which is not allowed", r.RemoteAddr, path)
+		http.Error(w, fmt.Sprintf("querying config value '%s' is not allowed", path), http.StatusForbidden)
 		return
 	}
 
-	w.WriteHeader(statusCode)
-	_, err = w.Write(body)
-	if err != nil {
-		log.Warnf("config endpoint: could not write response body: %v", err)
+	log.Debug("config endpoint received a request from '%s' for config '%s'", r.RemoteAddr, path)
+	value := c.cfg.Get(path)
+	if value == nil {
```

Comment:
A setting could be `nil` as `null` is a valid YAML type. If we want to handle this case we should use the `IsSet` method on the configuration.
But should we care ? If something it not set, returning `nil`/`null` is a valid behavior and not really and error, no ?

Reply:
For this specific change I kept the same behavior as in my previous PR, but thinking about it I agree `nil` could be a valid value so it doesn't really makes sense to ignore it, I'll update the PR

Reply:
I also removed the expvar about nil values since we consider it a "normal" value now